### PR TITLE
Added Progression

### PR DESCRIPTION
**Pull request by wolearyc**
I've finished implementing the Progression tool (#255), including tests and documentation. It passes pre-commit. 

It would be super to include this feature in v3! Feel free to play around with it and provide feedback before accepting the pull request.

### CODE REVIEW

File: .github/workflows/lock-threads.yml
Comment (wolearyc) [inline .github/workflows/lock-threads.yml:1]: not sure what this is ¯\_(ツ)_/¯

File: .github/workflows/stalebot.yml
Comment (wolearyc) [inline .github/workflows/stalebot.yml:1]: or this ¯\_(ツ)_/¯

**Comment by mortii** [comment]
Wow, I'm blown away, that is super cool!

I'll do a couple of things:
1. I'll massage the git history a little bit to get rid of the merge commits before rebasing it onto the v3 branch
2. I might make some small spacing adjustments and other tweaks to this layout::
Screenshot from 2024-07-21 12-40-48

Thank you so much!

File: .github/workflows/lock-threads.yml
Comment (mortii) [reply .github/workflows/lock-threads.yml:1]: np, it's just an artifact of merging in the main branch, I'll make it dissapear :+1:

**Comment by wolearyc** [comment]
Sweet, thanks!

**Pull request merged**